crates/rpc/rpc/src/eth/error.rs

```diff
@@ -27,7 +30,19 @@ pub(crate) enum EthApiError {
     Internal(#[from] reth_interfaces::Error),
 }
 
-impl_to_rpc_result!(EthApiError);
+impl From<EthApiError> for RpcError {
+    fn from(value: EthApiError) -> Self {
+        match value {
+            EthApiError::UnknownBlockNumber => {
+                RpcError::Call(CallError::Custom(ErrorCode::InvalidParams.into()))
+            }
+            EthApiError::Internal(_) => {
+                RpcError::Call(CallError::Custom(ErrorCode::InternalError.into()))
+            }
+            err => RpcError::Call(CallError::from_std_error(err)),
```

Comment:
this needs to be a bit more detailed, because we need to set code,error,data for some errors.

See:

https://github.com/paradigmxyz/reth/blob/92ff3f961dd381ccb3b3dda144e84036fd7de5d4/crates/rpc/rpc/src/result.rs#L107-L132

for example the default impl for some errors would call `internal_rpc_err(err.to_string())` which results in `code=INTERNAL_ERROR_CODE,message=err,data=None`

for some errors, like `UnknownBlock` we need to set custom errors.

we could also add a new rpc error function that accepts code+message.

See also https://github.com/vapory-legacy/wiki/blob/master/JSON-RPC-Error-Codes-Improvement-Proposal.md